server/gorillamux/handlers_test.go

```diff
@@ -0,0 +1,53 @@
+package gorillamux
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/SKF/go-rest-utility/server/gorillamux/problems"
+)
+
+func Test_MethodNotFoundHandler(t *testing.T) {
+	expectedRes := problems.MethodNotAllowed("GET", "PUT", "POST")
+
+	router := mux.NewRouter()
+	router.Name("testRouter").Methods("PUT", "POST")
+
+	ts := httptest.NewServer(MethodNotFoundHandler(router))
+	defer ts.Close()
+	res, err := http.Get(ts.URL)
+
+	require.NoError(t, err)
+
+	reader := res.Body
+	defer reader.Close()
+
+	actual := problems.MethodNotAllowedProblem{}
+
+	require.NoError(t, json.NewDecoder(reader).Decode(&actual))
+	assert.Equal(t, expectedRes, actual)
```

Comment:
We should assert the same as the users can expect. The type, HTTP status and any custom fields we added to the problem.